### PR TITLE
Show easy language link for German only (EXPOSUREAPP-4986)

### PR DESCRIPTION
**Pull request by chiljamgossow**
... and simplify layout and fragment

device-2021-02-09-190619
device-2021-02-09-190556

### CODE REVIEW

File: Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/onboarding/OnboardingFragmentTest.kt [hunk 1]
```diff
@@ -31,19 +28,6 @@ class OnboardingFragmentTest : BaseUITest() {
     @get:Rule
     val systemUIDemoModeRule = SystemUIDemoModeRule()
 
-    @Before
-    fun setUp() {
```
Comment (ralfgehrer) [inline Corona-Warn-App/src/androidTest/java/de/rki/coronawarnapp/ui/onboarding/OnboardingFragmentTest.kt:35]: Why was it removed?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/onboarding/OnboardingFragment.kt [hunk 2]
```diff
@@ -9,46 +9,47 @@ import androidx.fragment.app.Fragment
...
-                is OnboardingNavigationEvents.NavigateToEasyLanguageUrl ->
-                    startActivity(
-                        Intent(
-                            Intent.ACTION_VIEW,
-                            Uri.parse(getString(R.string.onboarding_tracing_easy_language_explanation_url))
-                        )
-                    )
+            onboardingButtonNext.setOnClickListener { goToOnboardingPrivacyFragment() }
+            // only show link for German
+            if (Locale.getDefault().language == "de") {
```
Comment (ralfgehrer) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/onboarding/OnboardingFragment.kt:28]: This check could be implemented in the VM instead. The VM could expose a boolean `onboardingEasyLanguageEnabled` and a test could be written by easily mocking the Locale.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/onboarding/OnboardingFragment.kt [hunk 3]
```diff
@@ -9,46 +9,47 @@ import androidx.fragment.app.Fragment
...
+                onboardingEasyLanguage.visibility = View.VISIBLE
+                onboardingEasyLanguage.setOnClickListener { openEasyLanguageLink() }
+            } else {
+                onboardingEasyLanguage.visibility = View.GONE
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/onboarding/OnboardingFragment.kt:32]: adding to Ralf's suggestion:
`onboardingEasyLanguage.isVisible = onboardingEasyLanguageEnabled`

File: Corona-Warn-App/src/main/res/layout/fragment_onboarding.xml [hunk 4]
```diff
@@ -11,62 +11,124 @@
...
-            app:layout_constraintTop_toTopOf="parent"
-            app:subtitle="@{@string/onboarding_subtitle}" />
+            app:layout_constraintTop_toTopOf="parent">
+
+            <androidx.constraintlayout.widget.ConstraintLayout
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:paddingBottom="@dimen/spacing_medium">
+
+                <ImageView
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/fragment_onboarding.xml:28]: `AppCompatImageView`?

File: Corona-Warn-App/src/main/res/layout/fragment_onboarding.xml [hunk 5]
```diff
@@ -11,62 +11,124 @@
...
+                    android:id="@+id/onboarding_illustration"
+                    android:layout_width="0dp"
+                    android:layout_height="wrap_content"
+                    android:contentDescription="@string/onboarding_illustration_description"
+                    android:focusable="true"
+                    android:src="@drawable/ic_illustration_together"
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/res/layout/fragment_onboarding.xml:34]: `app:srcCompat`

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/onboarding/OnboardingFragment.kt (hunk 2 above)
Comment (chiljamgossow) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/onboarding/OnboardingFragment.kt:28]: I will move to a function. This way I can write a unit test and don't need a view model.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/onboarding/OnboardingFragment.kt (hunk 3 above)
Comment (chiljamgossow) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/onboarding/OnboardingFragment.kt:33]: I moved the logic to a global function and added unit tests

**Review by mtwalli — DISMISSED** [review]
lgtm

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/onboarding/OnboardingFragment.kt [hunk 6]
```diff
@@ -5,52 +5,57 @@ import android.net.Uri
...
     }
 
+    private fun goToOnboardingPrivacyFragment() {
+        doNavigate(
+            OnboardingFragmentDirections
+                .actionOnboardingFragmentToOnboardingPrivacyFragment()
+        )
+    }
+
+    private fun openEasyLanguageLink() {
```
Comment (BMItter) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/onboarding/OnboardingFragment.kt:45]: We have a helper for showing links. Check ExternalActionHelper

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/onboarding/OnboardingFragment.kt [hunk 7]
```diff
@@ -5,52 +5,57 @@ import android.net.Uri
...
+    }
+
     override fun onResume() {
         super.onResume()
         binding.onboardingContainer.sendAccessibilityEvent(AccessibilityEvent.TYPE_ANNOUNCEMENT)
     }
 }
+
+@VisibleForTesting
+fun showEasyLanguageLink(): Boolean = Locale.getDefault().language == "de"
```
Comment (BMItter) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/onboarding/OnboardingFragment.kt:61]: You could check against Locale.German.language, if you want

**Review by BMItr — APPROVED** [review]

**Review by mtwalli — APPROVED** [review]

**Pull request merged**